src/ipa_pc/data_structures.rs

```diff
@@ -892,3 +927,463 @@ impl<F: PrimeField> CanonicalDeserialize for SuccinctCheckPolynomial<F> {
+    fn is_valid(&self) -> bool {
+        self.comm.is_valid()
+    }
+}
+
+impl<G: AffineCurve> Mul<G::ScalarField> for SegmentedCommitment<G> {
+    type Output = Self;
+
+    #[inline]
+    fn mul(self, other: G::ScalarField) -> Self {
+        SegmentedCommitment {
+            comm: self.comm.iter().map(|&c| {
+                c.into_projective().mul(&other).into_affine()
+            }).collect::<Vec<_>>()
+        }
+    }
+}
+
+impl<G: AffineCurve> AddAssign<Self> for SegmentedCommitment<G> {
+    #[inline]
+    fn add_assign(&mut self, other: Self) {
+        for i in 0..other.comm.len() {
+            if i == self.comm.len() {
+                self.comm.push(G::zero())
+            }
+            let mut sum = self.comm[i].into_projective();
+            sum.add_assign_mixed(&other.comm[i]);
+            self.comm[i] = sum.into_affine();
+        }
+    }
+}
+
+impl<G: AffineCurve> Neg for SegmentedCommitment<G> {
+    type Output = Self;
+
+    #[inline]
+    fn neg(self) -> Self {
+        SegmentedCommitment {
+            comm: self.comm.iter().map(|&c| {
+                -c
+            }).collect::<Vec<_>>()
+        }
+    }
+}
+
+/// `Randomness` hides the polynomial inside a commitment and is outputted by `InnerProductArg::commit`.
+#[derive(Derivative)]
+#[derivative(
+Default(bound = ""),
+Hash(bound = ""),
+Clone(bound = ""),
+Debug(bound = ""),
+PartialEq(bound = ""),
+Eq(bound = "")
+)]
+#[derive(CanonicalSerialize, CanonicalDeserialize)]
+pub struct SegmentedRandomness<G: AffineCurve> {
+    /// Randomness is some scalar field element.
+    pub r: Vec<G::ScalarField>,
+}
+
+impl<G: AffineCurve> PCRandomness<G> for SegmentedRandomness<G> {
+
+    #[inline]
+    fn empty() -> Self {
+        Self {
+            r: vec![]
+        }
+    }
+}
+
+impl<F: Field, G: AffineCurve<ScalarField = F>> LinearCombination<F> for SegmentedRandomness<G> {
+    type Item = SegmentedRandomness<G>;
+
+    fn from_lc(_items: Vec<Self::Item>, _base: F, _step: usize) -> Self::Item {
+        Self::Item::empty()
+    }
+}
+
+impl<G: AffineCurve> Mul<G::ScalarField> for SegmentedRandomness<G> {
+    type Output = Self;
+
+    #[inline]
+    fn mul(self, other: G::ScalarField) -> Self {
+        SegmentedRandomness {
+            r: self.r.iter().map(|&r| {
+                r * other
+            }).collect::<Vec<_>>()
+        }
+    }
+}
+
+impl<G: AffineCurve> AddAssign<Self> for SegmentedRandomness<G> {
+    #[inline]
+    fn add_assign(&mut self, other: Self) {
+        for i in 0..other.r.len() {
+            if i == self.r.len() {
+                self.r.push(G::ScalarField::zero())
+            }
+            self.r[i] += other.r[i];
+        }
+    }
+}
+
+
+impl<G: AffineCurve> Neg for SegmentedRandomness<G> {
+    type Output = Self;
+
+    #[inline]
+    fn neg(self) -> Self {
+        SegmentedRandomness {
+            r: self.r.iter().map(|&r| {
+                -r
+            }).collect::<Vec<_>>()
+        }
+    }
+}
```

Comment:
Same here: to build linear combinations all one actually just needs a the group trait. 
This code duplication actually brings me back to the idea having an LC trait/struct for a general group G: 
It defines formal LCs  and `from_lc()` using the the group operations of G. Then one just give a quick implementation of the group trait for
1. affine curves and their "vector extension" (as we already did), 
2. prime fields and polynomials over it.